fuzz/fuzz_targets/differential.rs

```diff
@@ -613,31 +620,43 @@ impl FuzzContext {
     }
 }
 
-fuzz_target!(|data: &[u8]| {
-    let Ok(mut smith_module) = arbitrary_exec_module(data) else {
-        return;
-    };
-    // Note: We cannot use built-in fuel metering of the different engines since that
-    //       would introduce unwanted non-determinism with respect to fuzz testing.
-    let Ok(_) = smith_module.ensure_termination(1_000 /* fuel */) else {
-        return;
-    };
-    let wasm = smith_module.to_bytes();
-    let Some(wasmi_register) = <WasmiRegister as DifferentialTarget>::setup(&wasm[..]) else {
-        return;
-    };
-    let Some(wasmi_stack) = <WasmiStack as DifferentialTarget>::setup(&wasm[..]) else {
-        panic!("wasmi (register) succeeded to create Context while wasmi (stack) failed");
-    };
-    let exports = wasmi_register.exports();
-    let mut context = FuzzContext {
-        wasm,
-        wasmi_register,
-        wasmi_stack,
-        exports,
-    };
-    context.run();
-});
+fn main() {
+    loop {
+        fuzz!(|seed: &[u8]| {
+            let mut unstructured = Unstructured::new(seed);
+            let Ok(mut smith_module) =
+                arbitrary_config(&mut unstructured).and_then(|mut config| {
+                    config.reference_types_enabled = false;
+                    config.tail_call_enabled = false;
```

Comment:
I wonder why we do not use `tail-calls` proposal in differential fuzzing. 🤔 Do you have an idea?

Reply:
A TranslationError gets thrown by WasmiStack when trying to make the module

https://github.com/wasmi-labs/wasmi/blob/708bef07206c8805db92e7f4513cb5b2c729d51e/fuzz/fuzz_targets/differential.rs#L216

``called `Result::unwrap()` on an `Err` value: Module(Translation(TranslationError { inner: Validate(BinaryReaderError { inner: BinaryReaderErrorInner { message: "tail calls support is not enabled", offset: 157, needed_hint: None } })
 }))``

Reply:
This is probably because `tail-calls` (and `extended-const`) are disabled by default in Wasmi v0.31 (`wasmi_stack`).